test/e2e/apimachinery/garbage_collector.go

```diff
@@ -488,6 +492,9 @@ var _ = SIGDescribe("Garbage collector", func() {
 		if e, a := int(*(rc.Spec.Replicas)), len(pods.Items); e != a {
 			framework.Failf("expect %d pods, got %d pods", e, a)
 		}
+		if err = e2epod.DeletePodsWithGracePeriod(clientSet, pods.Items, 0); err != nil {
+			framework.Failf("failed to delete pods: %v", err)
+		}
 		gatherMetrics(f)
```

Comment:
should gatherMetrics be first?

Reply:
Yeah, not sure... probably.  I also wonder if these cleanup deletes should fail the test or just log warnings.

Reply:
I think I might just change this PR to do the garbage_collector only for now, since that is the one with the most pods, and then make another pr to experiment on the other files.